File: data-structures/strings/palindrome_test/is_palindrome.py
def is_palindrome(string):
    str_length = len(string)

    for i in range(0, str_length):
        if(string[i] != string[str_length - i - 1]):
            return False

    return True


string = input("Enter the string: ")

if is_palindrome(string):
    print("The string is a palindrome")

else :
    print("The string is not a palindrome")

File: data-structures/strings/palindrome_test/is_palindrome.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>


bool
is_palindrome(const char *string)
{
    size_t string_length, i;

    string_length = strlen(string) - 1;

    for(i = 0; i < string_length; i++) {
        if(string[i] != string[string_length - i -1]) {
            return false;
        }
    }

    return true;
}


int
main(void)
{
    char string[99];

    printf("Enter the string: ");
    fgets(string, sizeof(string), stdin);

    if(is_palindrome(string)) {
        printf("The string is a palindrome.\n");
    } else {
        printf("The string is not a palindrome.\n");
    }

    return 0;
}